src/test/java/api/loans/ScheduledNoticesTests.java

```diff
@@ -0,0 +1,506 @@
+      .withDueDateEvent()
+      .withBeforeTiming(beforePeriod)
+      .recurring(beforeRecurringPeriod)
+      .sendInRealTime(true)
+      .create();
+    JsonObject uponAtDueDateNoticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(uponAtTemplateId)
+      .withDueDateEvent()
+      .withUponAtTiming()
+      .sendInRealTime(false)
+      .create();
+    JsonObject afterDueDateNoticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(afterTemplateId)
+      .withDueDateEvent()
+      .withAfterTiming(afterPeriod)
+      .recurring(afterRecurringPeriod)
+      .sendInRealTime(true)
+      .create();
+
+    NoticePolicyBuilder noticePolicy = new NoticePolicyBuilder()
+      .withName("Policy with due date notices")
+      .withLoanNotices(Arrays.asList(
+        beforeDueDateNoticeConfiguration,
+        uponAtDueDateNoticeConfiguration,
+        afterDueDateNoticeConfiguration));
+
+    LoanPolicyBuilder loanPolicy = new LoanPolicyBuilder()
+      .withName("Can Circulate Rolling for recall")
+      .rolling(Period.weeks(3))
+      .withRecallsMinimumGuaranteedLoanPeriod(Period.weeks(2));
+
+    useLoanPolicyAsFallback(
+      loanPoliciesFixture.create(loanPolicy).getId(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.create(noticePolicy).getId());
+
+    IndividualResource item = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource borrower = usersFixture.steve();
+
+    final IndividualResource loan = loansFixture.checkOutByBarcode(item, borrower);
+    loansFixture.checkOutByBarcode(itemsFixture.basedUponNod(), usersFixture.jessica());
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, hasSize(6));
+
+    requestsFixture.place(new RequestBuilder()
+      .open()
+      .recall()
+      .forItem(item)
+      .by(usersFixture.james())
+      .withPickupServicePointId(servicePointsFixture.cd1().getId()));
+    IndividualResource loanAfterRecall = loansClient.get(loan.getId());
+    DateTime dueDateAfterRecall = getDateTimeProperty(loanAfterRecall.getJson(), "dueDate");
+
+    Matcher<Iterable<JsonObject>> scheduledNoticesAfterRecallMatcher = hasItems(
+      hasScheduledLoanNotice(
+        loan.getId(), dueDateAfterRecall.minus(beforePeriod.timePeriod()),
+        BEFORE_TIMING, beforeTemplateId,
+        beforeRecurringPeriod, true),
+      hasScheduledLoanNotice(
+        loan.getId(), dueDateAfterRecall,
+        UPON_AT_TIMING, uponAtTemplateId,
+        null, false),
+      hasScheduledLoanNotice(
+        loan.getId(), dueDateAfterRecall.plus(afterPeriod.timePeriod()),
+        AFTER_TIMING, afterTemplateId,
+        afterRecurringPeriod, true)
+    );
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticesClient::getAll, scheduledNoticesAfterRecallMatcher);
+    assertThat(scheduledNoticesClient.getAll(), hasSize(6));
```

Comment:
What happens if the wrong 6 notices are scheduled?

Reply:
I skipped this check as it is verified in tests for scheduling notices on checkout.
So if wrong notices are scheduled, other tests will demonstrate it. 
What do you think?

Reply:
Could there be different notices applied, between a check out and renewal?

Reply:
Yes it possible case.

Reply:
Sorry, I thought you mean size check after checkout...
This should check, that rescheduled notices are correct.
` .until(scheduledNoticesClient::getAll, scheduledNoticesAfterRecallMatcher);`

Reply:
Ok, if that checks that all of the correct notices are present, why check the count afterwards?

Reply:
`hasItems()` checks that collection contains particular notices (notices scheduled based on new due date). But it doesn't check case when redundant notices appear ( for example that old notices weren't removed). So to check all the cases, it should check expected rescheduled notices, and recheck collection size.

Reply:
Ok, thank you for guiding me through it

Reply:
Does the code need to accommodate a case where the rules or the request policy have changed in between check out and renewal?

Is there a test to demonstrate that?

Reply:
I came to the conclusion that the functionality of scheduling notices based on loan and notice policy is already covered with tests, and there is no special condition for the case when notice policy is changed (as notices will be scheduled in the same way). I think we'll need this kind of test when we have some logic for changed notice policy.
What do you think?

Reply:
Personally, I think it is a meaningful scenario. I shall leave the decision to you, as you are more familiar with the intended behaviour of the system than I am.